wandb/sdk/artifacts/artifact.py

```diff
@@ -501,6 +480,57 @@ def metadata(self, metadata: dict) -> None:
         """
         self._metadata = self._normalize_metadata(metadata)
 
+    @property
+    def ttl(self) -> Union[datetime.timedelta, None]:
+        """Artifact TTL(time to live).
+
+        Set a TTL to mark the artifact for deletion after the given duration from Artifact creation has passed.
+        To turn off previously set artifact TTL, reset TTL to wandb.ArtifactTTL.DEFAULT.
+
+        Examples:
+            Basic usage
+            ```
+            artifact.ttl = timedelta(days=10) # artifact will be marked for deletion on artifact.created_at + 10 days
+            artifact.save() # or log the artifact if its new
+            ```
+        """
+        if self._ttl_change == ArtifactTTLChange.INHERITED:
+            raise Exception(
+                "Unable to fetch Inherited TTL until artifact is logged/saved"
+            )
+        if self._ttl_duration_seconds is None:
+            return None
+        return datetime.timedelta(seconds=self._ttl_duration_seconds)
+
+    @ttl.setter
+    def ttl(self, ttl: Union[datetime.timedelta, ArtifactTTL, None]) -> None:
+        """Artifact TTL(time to live).
+
+        Note:
+            This ttl is set on the artifact version level.
+
+        Arguments:
+            ttl: How long the artifact will remain active from its creation. timedelta must be positive,
+            To reset artifact TTL to inherit from collection level rules `artifact.ttl = wandb.ArtifactTTL.INHERIT`
+        """
+        if self.type == "wandb-history":
+            raise ValueError("Cannot set artifact TTL for type wandb-history")
+
```

Comment:
...then we should always set
```suggestion
        self._ttl_changed = True
```